pkg/virt-controller/watch/vmi.go

```diff
@@ -1288,9 +1298,18 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 		}
 
 		if vmiSpecIfaces, vmiSpecNets, dynamicIfacesExist := network.CalculateInterfacesAndNetworksForMultusAnnotationUpdate(vmi); dynamicIfacesExist {
-			if err := c.updateMultusAnnotation(vmi.Namespace, vmiSpecIfaces, vmiSpecNets, pod); err != nil {
+			errMessage := fmt.Errorf("failed to hot{un}plug network interfaces for vmi [%s/%s]", vmi.GetNamespace(), vmi.GetName())
+
+			networkToIPAMClaimParams, err := network.GetNetworkToIPAMClaimParams(c.clientset, vmi.Namespace, vmi.Name, vmiSpecNets)
+			if err != nil {
+				return &syncErrorImpl{
+					err:    fmt.Errorf("%s: %w", errMessage, err),
+					reason: FailedHotplugSyncReason,
+				}
+			}
+			if err := c.updateMultusAnnotation(vmi.Namespace, vmiSpecIfaces, vmiSpecNets, pod, networkToIPAMClaimParams); err != nil {
```

Comment:
why are we handling hotplug in this PR ? Didn't we agree on doing that as a follow up ?

Reply:
It is not about hot plug in our case.
`CalculateInterfacesAndNetworksForMultusAnnotationUpdate` checks if the status is different from spec
so even if we start a VMI with all the interfaces already there, as long as the status is different from the spec, we will reach here in reconcile, and then if we don't add here the parameters, the pod annotation will be removed